diff -ruN lzma-4.06./SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp lzma-4.06/SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp
--- lzma-4.06./SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp	2004-09-05 21:35:02.000000000 +0200
+++ lzma-4.06/SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp	2004-10-21 22:58:06.377959392 +0200
@@ -129,10 +129,10 @@
 
 int main2(int n, const char *args[])
 {
-  fprintf(stderr, "\nLZMA 4.06 Copyright (c) 1999-2004 Igor Pavlov  2004-09-05\n");
-
   if (n == 1)
   {
+    fprintf(stderr, "\nLZMA 4.06 Copyright (c) 1999-2004 Igor Pavlov  2004-09-05\n");
+
     PrintHelp();
     return 0;
   }
