diff -ruN lzma-4.42./C/7zip/Compress/Branch/BranchTypes.h lzma-4.42/C/7zip/Compress/Branch/BranchTypes.h
--- lzma-4.42./C/7zip/Compress/Branch/BranchTypes.h	2006-04-22 08:34:37.000000000 +0200
+++ lzma-4.42/C/7zip/Compress/Branch/BranchTypes.h	2006-05-27 20:43:03.420392912 +0200
@@ -3,13 +3,6 @@
 #ifndef __BRANCHTYPES_H
 #define __BRANCHTYPES_H
 
-typedef unsigned char Byte;
-typedef unsigned short UInt16;
-
-#ifdef _LZMA_UINT32_IS_ULONG
-typedef unsigned long UInt32;
-#else
-typedef unsigned int UInt32;
-#endif
+#include "../LZMA_C/LzmaTypes.h"
 
 #endif
