--- ./C/7zip/Compress/LZMA_Lib/makefile~	2006-10-30 22:54:27.135579277 +0200
+++ ./C/7zip/Compress/LZMA_Lib/makefile	2006-10-30 22:54:28.415608066 +0200
@@ -25,11 +25,14 @@
   Vector.o \
 
 
-all: $(PROG)
+all: $(PROG) liblzma.so.0.0.0
 
 $(PROG): $(OBJS)
 	$(AR) r $(PROG) $(OBJS)
 
+liblzma.so.0.0.0:	$(OBJS)
+	$(CXX) -shared -o $@ -Wl,-soname,liblzma.so.0 $(OBJS)
+
 ZLib.o: ZLib.cpp
 	$(CXX) $(CFLAGS) ZLib.cpp
 
